packages/styled-components/src/constructors/css.ts

```diff
@@ -18,10 +18,18 @@ const addTag = <T>(arg: T extends any[] ? T & { isCss?: boolean } : T) => {
   return arg;
 };
 
-export default function css<Props extends object>(
+function css(
+  styles: Styles<{}>,
+  ...interpolations: Interpolation<{}>[]
+): Interpolation<{}>;
+function css<Props extends object>(
   styles: Styles<Props>,
   ...interpolations: Interpolation<Props>[]
-) {
+): Interpolation<Props>;
+function css<Props extends {} = {}>(
+  styles: Styles<Props>,
+  ...interpolations: Interpolation<Props>[]
+): Interpolation<Props> {
   if (isFunction(styles) || isPlainObject(styles)) {
```

Comment:
This seems like the crux of the fix, because the `css` function is most often called without explicit `<Props>` and should not try to infer `Props` from interpolations in that case. Here is the variant I've been testing:

```ts
function css(
  styles: Styles<object>,
  ...interpolations: Interpolation<object>[]
): Interpolation<object>;
function css<Props extends object>(
  styles: Styles<Props>,
  ...interpolations: Interpolation<Props>[]
): Interpolation<Props>;
function css<Props extends object = object>(
  styles: Styles<Props>,
  ...interpolations: Interpolation<Props>[]
) {
```

Would you mind changing yours to match that, especially using `object` instead of `{}`?